Alpaca.Markets/PolygonSockClient.cs

```diff
@@ -0,0 +1,274 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using WebSocket4Net;
+using System.Security.Authentication;
+
+namespace Alpaca.Markets
+{
+    /// <summary>
+    /// Provides unified type-safe access for Polygon streaming API via websockets.
+    /// </summary>
+    public sealed partial class PolygonSockClient : SockClientBase
+    {
+        private readonly String _keyId;
+
+        /// <summary>
+        /// Occured when new trade received from stream.
+        /// </summary>
+        public event Action<IStreamTrade> TradeReceived;
+
+        /// <summary>
+        /// Occured when new quote received from stream.
+        /// </summary>
+        public event Action<IStreamQuote> QuoteReceived;
+
+        /// <summary>
+        /// Occured when new bar received from stream.
+        /// </summary>
+        public event Action<IStreamAgg> MinuteAggReceived;
+
+        /// <summary>
+        /// Occured when new bar received from stream.
+        /// </summary>
+        public event Action<IStreamAgg> SecondAggReceived;
+
+        /// <summary>
+        /// Occured when any error happened in stream.
+        /// </summary>
+        public event Action<Exception> OnError;
```

Comment:
We can move `OnError` event into base class `SockClientBase` and implement error forwarding in this class and only once.

Reply:
@OlegRa could you give some clarification on how to move it to the base class? That was my instinct as well, but when I attempted to do it, I got an error saying that an event can't be used in a subclass except on the left side of += or -=. I'm not sure what the rest of your comment refers to, so maybe that's the workaround.

Reply:
I'll do it in separate commit after merge. It's not critical now.